api/v1beta1/azuremachine_webhook_test.go

```diff
@@ -145,6 +145,21 @@ func TestAzureMachine_ValidateCreate(t *testing.T) {
 			machine: createMachineWithNetworkConfig("", nil, []NetworkInterface{{SubnetName: "subnet", PrivateIPConfigs: 1}}),
 			wantErr: false,
 		},
+		{
+			name:    "azuremachine without confidential compute properties and encryption at host enabled",
+			machine: createMachineWithConfidentialCompute(true, "", "", ""),
+			wantErr: false,
+		},
+		{
+			name:    "azuremachine with valid security profile with confidential VM properties enabled",
+			machine: createMachineWithConfidentialCompute(false, SecurityEncryptionTypeVMGuestStateOnly, VirtualizedTrustedPlatformModulePolicyEnabled, SecureBootPolicyEnabled),
+			wantErr: false,
+		},
+		{
+			name:    "azuremachine with confidential compute properties and encryption at host enabled",
+			machine: createMachineWithConfidentialCompute(true, SecurityEncryptionTypeVMGuestStateOnly, "", ""),
+			wantErr: true,
+		},
```

Comment:
Are you able to add some more test cases here that reflect the cases in `azuremachine_validation_test.go` such as with DiskWithVMGuestState encryption and vTPM disabled?

Reply:
Yes, it absolutely makes sense to be as comprehensive as possible here. I believe I added the test cases that were missing, but please let me know if you have thought about additional ones.